### ISSUE TITLE
Unable to select Team/Provider when running eas build

### ISSUE THREAD

**Issue by oalsing** [expo/eas-cli#1863]
### Summary

Same issue as reported in #1731.

I'm unable to select another team and provider when running eas build, even when setting the environment variables as per the suggested solution.

              Please use the EXPO_APPLE_TEAM_ID environment variable to configure the team, eg https://github.com/expo/eas-cli/blob/95155681db4ed7666bf7b5a7c5f9873625ac9b17/packages/eas-cli/src/credentials/ios/appstore/__tests__/resolveCredentials-test.ts#L63

_Originally posted by @EvanBacon in https://github.com/expo/eas-cli/issues/1731#issuecomment-1468371377_
            

Regardless of the value of the EXPO_APPLE_TEAM_ID, my previously used Team Id is always selected per default, and I'm thus unable to select the team/provider for the App ID I'm building.

### Managed or bare?

managed

### Environment

npx expo-doctor

### Error output

✔ Validating global prerequisites versions passed
✔ Checking for incompatible packages passed
✔ Checking for conflicting global packages in project passed
✔ Verifying prebuild support package versions are compatible passed
✔ Checking dependency versions for compatibility with the installed Expo SDK passed
✔ Validating Expo Config passed
✖ Checking package.json for common issues failed
Issues:
  The following scripts in package.json conflict with the contents of node_modules/.bin: eslint, lint-staged, prettier.

✖ Found one or more possible issues with the project. See above logs for issues and advice to resolve.

### Reproducible demo or steps to reproduce from a blank project

Running eas build for any project that belongs to a different team than the previous project.

**Comment by oalsing** [comment]
@szdziedzic 

```sh
EXPO_APPLE_TEAM_ID=XYZ eas build --platform ios
```

Where `XYZ` is the actual team name that I want to use.

The team is automatically selected and I'm unable to select any other team.

**Comment by szdziedzic** [comment]
Could you check if the `~/.app-store` directory exists on your device and if so try to delete it? This will clean all App Store related cookies used by the CLI from your device. After this is done you can try to authenticate again and see if it resolves your issue.

**Comment by oalsing** [comment]
Unfortunately, that did not allow me to select another team, it was still selected automatically. However, I was able to select the provider, which was selected automatically previously.

**Comment by MetalWeirdo** [comment]
I'm also experiencing the same issue. 
I pick a team but the other team is always selected.

I've removed all data in `.app-store` and re-authenticated but still experience the same issue. 

This is pretty blocking as I can't build my app with the right team 😅 

```bash
➜ eas --version 
eas-cli/3.13.3 darwin-arm64 node-v16.14.2
➜ eas build --platform ios
✔ Using remote iOS credentials (Expo server)

If you provide your Apple account credentials we will be able to generate all necessary build credentials and fully validate them.
This is optional, but without Apple account access you will need to provide all the missing values manually and we can only run minimal validation on them.
✔ Do you want to log in to your Apple account? … yes

› Log in to your Apple Developer account to continue
✔ Apple ID: … xxxxxxxxxxxxxxx
› Using password for xxxxxxxxxxxxxxx from your local Keychain
  Learn more
✔ Logged in, verify your Apple account to continue
Two-factor Authentication (6 digit code) is enabled for xxxxxxxxxxxxxxx. Learn more.

✔ How do you want to validate your account? … device / sms
✔ Please enter the 6 digit code … xxxxxx
✔ Valid code
✔ Logged in and verified
› Team xxxxxxxxxxxxxxx (xxxxxxxxxxxxxxx)
✔ Select a Provider › yyyyyyyy (yyyyyyyyy)
✖ The bundle identifier com.foo.bar is not available to team "xxxxxxxxxxxxxxx (Company/Organization)" (xxxxxxxxxxxxxxx), change it in your app config and try again.
An attribute in the provided entity has invalid value - An App ID with Identifier 'com.foo.bar' is not available. Please enter a different string.
    Error: build command failed.
```

**Comment by andreiraduglavan** [comment]
I am experiencing the same issue. This is a real problem as I am building an app for a non-programmer client, that doesn't have the necessary skills to make a build on his own.

**Comment by oalsing** [comment]
@EvanBacon any updates on this?

**Comment by jorgerojas26** [comment]
I'm having the same issue, this is a real block for us.

**Comment by lorenzpfei** [comment]
It seems that the issue still persists. Any updates or workarounds on this?

**Comment by amaranth** [comment]
This variable appears to only be used when you're using an API key, not when you're doing user/pass login. The user/pass login flow does have the ability to override the team but as far as I can tell it's only used for `eas submit`.

**Comment by mojavad** [comment]
Yep, have confirmed it only works for `eas submit` on my side too.

**Comment by beydogan** [comment]
Having the same issue. One team comes auto selected and cannot override. It is a blocker for us too.

**Comment by AlexandraOlegovna** [comment]
> It seems that the issue still persists. Any updates or workarounds on this?
> 
> Workaround (edit):
> 
> * I successfully managed to deploy it by simply creating a new Apple account, adding it to the organisation and then logging in with it. For sure this is not the best solution but might help you for now.

Thank you! It works!

I took the following steps:
1. logged in to the Apple Developer Portal and left the team
2. asked another member to add my account to the team again
3. and now this team appears when I run `eas build` or `eas credentials`

**Comment by brad-ps** [comment]
Same issue. Trying the second developer account workaround.

**Comment by codergautam** [comment]
If I make a new account I get the error:

Authentication with Apple Developer Portal failed!
Apple provided the following error info:
You are not registered as an Apple Developer.  Please visit Apple Developer Registration. https://developer.apple.com/register/

**Comment by codergautam** [comment]
Dont want to pay 100 again

**Comment by codergautam** [comment]
After several hours this is what I found (thanks to GeyserJolt943 on Github)

Basically I was building a project for a non-technical client and needed to upload their app in their team. They added me as admin with sufficient permissions but it would not work.

It comes down to this:

> Note: If you’re enrolled as an individual and add users in App Store Connect, users receive access only to your content in App Store Connect and are not considered part of your team in the Apple Developer Program.

My client had a individual account, not organization so that means you only get access to the app not the bundle IDs. This gave me the following error:

```
✖ The bundle identifier ai.barbosa.app is not available to team XXX, change it in your app config and try again.
An attribute in the provided entity has invalid value - An App ID with Identifier XXX is not available. Please enter a different string.
```

Hope this helps someone 👍

**Comment by nizar0x1f** [comment]
The EAS CLI should always prompt for team selection. This issue seems to be stale.

**Comment by MehmetKaplan** [comment]
I am a bit confused how `eas build` and `eas submit` behave. Seems as if they behave different from the "provider" perspective.

In my scenario, I am developing an app for a provider (my customer) and also I am set as a developer for that provider.

- For the `eas build`, I need to build the app by choosing the provider. And in order to do that I need to remove the cookie each time so that the build may ask me which provider I choose "this time", or else the command tries to build as if I am the provider, not the customer:
   ```bash
   rm /Users/mehmetkaplan/.app-store/auth/[MY-DEVELOPER-ACCOUNT-EMAIL]/cookie
   ```
- For the `eas submit`, I need to submit on behalf of my developer account, if I remove the cookie again and try to submit by choosing the customer as the provider I got the below error (which I doubt if it is really a momentary Apple server error or a wrong-doing that spits out wrong error):
   ```txt
   Internal Server Error: 'h', 'Received an internal server error from Apple's App Store Connect / Developer Portal servers, please try again later.
   
   An unexpected error occurred. - An unexpected error occurred on the server side. If this issue continues, contact us at https://developer.apple.com/contact/.'. Retrying after 3 seconds (remaining: 3)
   
   ✖ Failed to create App Store app [MY APP]
   Received an internal server error from Apple's App Store Connect / Developer Portal servers, please try again later.
   
   An unexpected error occurred. - An unexpected error occurred on the server side. If this issue continues, contact us at https://developer.apple.com/contact/.
   Visit https://appstoreconnect.apple.com and resolve any warnings, then try again.
   Submission failed
       Error: submit command failed.
   ```

**Comment by keaton569** [comment]
i fixed this by closing my terminals and then opening them again.
